1.4/wheezy/Dockerfile

```diff
@@ -7,17 +7,19 @@ RUN apt-get update && apt-get install -y \
 	&& rm -rf /var/lib/apt/lists/*
 
 ENV GOLANG_VERSION 1.4.2
+ENV GOLANG_DOWNLOAD_URL https://golang.org/dl/go$GOLANG_VERSION.linux-amd64.tar.gz
+ENV GOLANG_DOWNLOAD_SHA1 6fede71082a2f57c13f2521ecd054b86ae77d0e5
 
-RUN curl -sSL https://golang.org/dl/go$GOLANG_VERSION.src.tar.gz \
-		| tar -v -C /usr/src -xz
+RUN mkdir -p /usr/src/golang \
+	&& curl -sSL "$GOLANG_DOWNLOAD_URL" -o golang.tar.gz \
+	&& echo "$GOLANG_DOWNLOAD_SHA1 golang.tar.gz" | sha1sum -c - \
```

Comment:
NEVERMIND, turns out, it requires literally two spaces between `<checksum>  <file>`. Not just 1. Silly me.